oml/functional/knn.py

```diff
@@ -1,55 +1,59 @@
 from typing import Optional, Sequence, Tuple
 
 import torch
-from torch import BoolTensor, FloatTensor, LongTensor
+from torch import BoolTensor, FloatTensor, LongTensor, Tensor
 from tqdm.auto import tqdm
 
 from oml.const import BS_KNN
 from oml.utils.misc_torch import pairwise_dist
 
 
-def batched_knn(
-    embeddings: FloatTensor,
-    ids_query: LongTensor,
-    ids_gallery: LongTensor,
+def check_both_are_nones_or_not_nones(a: Optional[Tensor], b: Optional[Tensor]) -> bool:
+    return (a is not None and b is not None) or (a is None and b is None)
+
+
+def check_len_and_ndim(a: Optional[Tensor], expected_len: int, expected_ndim: int) -> bool:
+    return (a is None) or (len(a) == expected_len and a.ndim == expected_ndim)
+
+
+def batched_knn_qg(
+    embeddings_query: FloatTensor,
+    embeddings_gallery: FloatTensor,
     top_n: int,
-    sequence_ids: Optional[LongTensor] = None,
-    labels_gt: Optional[LongTensor] = None,
+    labels_query: Optional[LongTensor] = None,
+    labels_gallery: Optional[LongTensor] = None,
+    ids_query: Optional[LongTensor] = None,
+    ids_gallery: Optional[LongTensor] = None,
+    sequence_ids_query: Optional[LongTensor] = None,
+    sequence_ids_gallery: Optional[LongTensor] = None,
     bs: int = BS_KNN,
     verbose: bool = False,
 ) -> Tuple[Sequence[FloatTensor], Sequence[LongTensor], Optional[Sequence[LongTensor]]]:
     """
+    Arguments have the same meaning as in `batched_knn`.
```

Comment:
I believe here we need to describe arguments anyway. We shouldn't expect from user to know api of different method.

Reply:
or mark it with underscore as a private method

Reply:
discussed offline: it's okay for internal function